ui/src/admin/actions/index.js

```diff
@@ -306,25 +308,29 @@ export const killQueryAsync = (source, queryID) => (dispatch) => {
   killQueryProxy(source, queryID)
 }
 
-export const deleteRoleAsync = (role, addFlashMessage) => (dispatch) => {
-  // optimistic update
+export const deleteRoleAsync = (role) => async (dispatch) => {
   dispatch(deleteRole(role))
-
-  // delete role on server
-  deleteRoleAJAX(role.links.self, addFlashMessage, role.name)
+  dispatch(publishAutoDismissingNotification('success', 'Role deleted'))
+  try {
+    await deleteRoleAJAX(role.links.self)
```

Comment:
yeah, good call. submitted the changes. the thought was that there's a trade-off. currently we're performing optimistic updates (aka updating state as though AJAX will be successful), which makes when to display the notification trickier: either 1) it notifies success at the same time as deletion of the item in the UI, but then an error message could come up shortly thereafter, which is confusing; or, 2) it removes the item from the UI, and notifies success upon AJAX success, which prevents showing both success and failure notifications, but also introduces a delay between UI state change and when the notification will present. thoughts?